java/src/jmri/jmrix/cmri/serial/serialmon/SerialMonFrame.java

```diff
@@ -4,20 +4,78 @@
 import jmri.jmrix.cmri.serial.SerialReply;
+import jmri.jmrix.cmri.serial.SerialNode;
+
+import java.awt.Dimension;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import javax.swing.JFileChooser;
+import jmri.util.FileUtil;
+import java.io.PrintStream;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+
+import javax.swing.BoxLayout;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.border.Border;
+import javax.swing.border.TitledBorder;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import javax.swing.text.BadLocationException;
 
 /**
  * Frame displaying (and logging) CMRI serial command messages
  *
  * @author Bob Jacobsen Copyright (C) 2001
+ * @author Chuck Catania  Copyright (C) 2014, 2016, 2017
  */
 public class SerialMonFrame extends jmri.jmrix.AbstractMonFrame implements SerialListener {
+    // member declarations
+    public boolean loggingEnabled = false;  // true if message logging to a file
+    public boolean freezeDisplay  = false;  // true if display is halted
+
+    protected JButton doneButton = new JButton();
+    protected JButton logMsgButton = new JButton();
+    protected JCheckBox deltaTBox = new JCheckBox();
+    protected JButton packetfilterButton = new JButton();  //c2
+    String rawDataCheck = this.getClass().getName()+".RawData";
+    String timeStampCheck = this.getClass().getName()+".TimeStamp";
+    String deltaTCheck = this.getClass().getName()+".DeltaT";
+    String alwaysOnTopCheck = this.getClass().getName()+".alwaysOnTop";
+    String autoScrollCheck = this.getClass().getName()+".AutoScroll";
+    jmri.UserPreferencesManager p;
+    
+    protected long lastTicks = 0L;
+    protected static int _DLE    = 0x10;    
+
+    final javax.swing.JFileChooser logFileChooser = new JFileChooser(FileUtil.getHomePath()); //jmri.jmrit.XmlFile.userFileLocationDefault());
+//    final javax.swing.JFileChooser logFileChooser = new JFileChooser(FileUtil.getUserFilesPath()); //jmri.jmrit.XmlFile.userFileLocationDefault());
 
     private CMRISystemConnectionMemo _memo = null;
 
     public SerialMonFrame(CMRISystemConnectionMemo memo) {
         super();
         _memo = memo;
     }
+    
+    @Override
+    public void dispose() { 
+        p.setSimplePreferenceState(timeStampCheck, timeCheckBox.isSelected());
+        p.setSimplePreferenceState(rawDataCheck, rawCheckBox.isSelected());
+        p.setSimplePreferenceState(alwaysOnTopCheck, alwaysOnTopCheckBox.isSelected());
+        p.setSimplePreferenceState(autoScrollCheck, !autoScrollCheckBox.isSelected());
+        _memo.getTrafficController().removeSerialListener(this);
+    }
```

Comment:
You must *always* call super.dispose() in a dispose method to ensure that all resources in the frame are disposed.

Reply:
Initially super.dispose() was called.  When exiting PanelPro an uncaught exception was thrown.
[Uncaught Exception Log.txt](https://github.com/JMRI/JMRI/files/1326613/Uncaught.Exception.Log.txt)
Without the call, the app exits cleanly.

initComponents in SerialMonFrame has additional items which are not in the Abstract class.  @Override is done on initComponents.
What is the correct approach for cleaning up in this instance?